### ISSUE TITLE
0!=0

### ISSUE THREAD

**Issue by ohhmm** [boostorg/multiprecision#43]
try this test
    boost::multiprecision::cpp_int i = 0;
    BOOST_TEST(i==0);
    ++i--;
    BOOST_TEST(i==0);
it fails in 1.65 because operator== compares limbs count.

**Comment by jzmaddock** [comment]
Yup, eval_increment/decrement don't set the sign to zero when the result yields zero, testing a fix now.
